src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/services/TaskManagementServiceTest.java

```diff
@@ -2994,5 +3000,131 @@ void should_throw_resource_not_found_exception_when_task_id_not_present() {
 
     }
 
+    @Nested
+    @DisplayName("getTaskRolePermissions()")
+    class GetTaskRolePermissions {
+
+        private final TaskRolePermissions mockedMappedTask = mock(TaskRolePermissions.class);
+
+        @Test
+        void should_succeed() {
+            String taskId = "taskId";
+            AccessControlResponse accessControlResponse = mock(AccessControlResponse.class);
+            TaskResource taskResource = spy(TaskResource.class);
+            TaskRoleResource tribunalResource = new TaskRoleResource(
+                "tribunal-caseworker", true, true, true, true, true,
+                true, new String[]{"Divorce"}, 1, false, "LegalOperations"
+            );
+
+            TaskRoleResource caseMgresource = new TaskRoleResource(
+                "case-manager", true, true, true, true, true,
+                true, new String[]{"Divorce"}, 1, false, "roleCategory"
+            );
+
+            Set<TaskRoleResource> taskRoleResourceSet = Set.of(tribunalResource, caseMgresource);
+            when(Optional.of(taskResource).get().getTaskRoleResources()).thenReturn(taskRoleResourceSet);
+            when(cftTaskDatabaseService.findByIdOnly(taskId)).thenReturn(Optional.of(taskResource));
+            when(cftTaskDatabaseService.findTaskBySpecification(any())).thenReturn(Optional.of(taskResource));
+
+            final List<TaskRolePermissions> taskRolePermissions = taskManagementService.getTaskRolePermissions(
+                taskId, accessControlResponse);
+
+            assertNotNull(taskRolePermissions);
+            assertFalse(taskRolePermissions.isEmpty());
+
+            assertTrue(taskRolePermissions.size() == 2);
```

Comment:
assertEquals()?

Reply:
Makes sense assertEquals gives more information when test fails. Will change it in RWA-810

Reply:
👍